pulsar-io/netty/src/main/java/org/apache/pulsar/io/netty/server/NettyServer.java

```diff
@@ -79,16 +84,45 @@ public void shutdownGracefully() {
             bossGroup.shutdownGracefully();
     }
 
+    private void runUdp() throws InterruptedException {
+        Bootstrap bootstrap = new Bootstrap();
+        bootstrap.group(workerGroup);
+        bootstrap.channel(NioDatagramChannel.class);
+        bootstrap.handler(new NettyChannelInitializer(new NettyServerHandler(this.nettySource)))
+                .option(ChannelOption.SO_BACKLOG, 1024);
+
+        ChannelFuture channelFuture = bootstrap.bind(this.host, this.port).sync();
+        channelFuture.channel().closeFuture().sync();
+    }
+
+    private void runTcp() throws InterruptedException {
+        ServerBootstrap serverBootstrap = new ServerBootstrap();
+        serverBootstrap.group(bossGroup, workerGroup);
+        serverBootstrap.channel(NioServerSocketChannel.class);
+        serverBootstrap.childHandler(new NettyChannelInitializer(new NettyServerHandler(this.nettySource)))
+                .option(ChannelOption.SO_BACKLOG, 1024)
+                .childOption(ChannelOption.SO_KEEPALIVE, true);
+
+        ChannelFuture channelFuture = serverBootstrap.bind(this.host, this.port).sync();
+        channelFuture.channel().closeFuture().sync();
+    }
+
     /**
      * Pulsar Tcp Server Builder.
      */
     public static class Builder {
 
+        private String type;
         private String host;
         private int port;
-        private NettyTcpSource nettyTcpSource;
+        private NettySource nettySource;
         private int numberOfThreads;
 
+        public Builder setType(String type) {
```

Comment:
We can support type-safety by defining `enum NettyServerType` instead of String.